client/galaxy/scripts/components/citations.vue

```diff
@@ -0,0 +1,196 @@
+                <p v-html="formattedReferences">
+                </p>
+            </span>
+            <textarea v-else style="width: 100%; height: 500px;" class="citations-bibtex-text">
+                {{ content }}
+            </textarea>
+        </div>
+    </div>
+</template>
+<script>
+import axios from "axios";
+import * as bibtexParse from "libs/bibtexParse";
+import { convertLaTeX } from "latex-to-unicode-converter";
+import { stringifyLaTeX } from "latex-parser";
+
+export default {
+    props: {
+        source: {
+            type: String,
+            required: true
+        },
+        id: {
+            type: String,
+            required: true
+        },
+        viewRender: {
+            type: Boolean,
+            requried: false,
+            default: true
+        }
+    },
+    data() {
+        return {
+            citations: [],
+            content: "",
+            errors: []
+        };
+    },
+    computed: {
+        formattedReferences: function() {
+            return this.citations.reduce((a, b) => a.concat(`<p>${this.formattedReference(b)}</p>`), "");
+        }
+    },
+    created: function() {
+        axios
+            .get(`${Galaxy.root}api/${this.source}/${this.id}/citations`)
+            .then(response => {
+                this.content = "";
+                for (var rawCitation of response.data) {
+                    try {
+                        var citation = {
+                            fields: {},
+                            entryType: undefined
+                        };
+                        var parsed = bibtexParse.toJSON(rawCitation.content);
+                        if (parsed) {
+                            parsed = _.first(parsed);
+                            citation.entryType = parsed.entryType || undefined;
+                            for (var key in parsed.entryTags) {
+                                citation.fields[key.toLowerCase()] = parsed.entryTags[key];
+                            }
+                        }
+                        this.citations.push(citation);
+                        this.content += rawCitation.content;
+                    } catch (err) {
+                        console.warn("Error parsing bibtex: " + err);
+                    }
+                }
+            })
+            .catch(e => {
+                console.error(e);
+            });
+    },
+    methods: {
+        formattedReference: function(citation) {
+            var entryType = citation.entryType;
+            var fields = citation.fields;
+
+            var ref = "";
+            var authorsAndYear = `${this._asSentence(
+                (fields.author ? fields.author : "") + (fields.year ? ` (${fields.year})` : "")
+            )} `;
+            var title = fields.title || "";
+            var pages = fields.pages ? `pp. ${fields.pages}` : "";
+            var address = fields.address;
+            if (entryType == "article") {
+                var volume =
```

Comment:
We can use `let` for block-level variables instead of `var` as we have started using ES6 features.

Reply:
Yep, we should.  Same comment as above; right now this is pulled straight from the old backbone classes and needs more updates.